src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -98,6 +98,24 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         codeWriter.emitTypeVariables(typeVariables)
 
         primaryConstructor?.let {
+          var useKeyword = false
+          val emittedAnnotations = if (it.annotations.isNotEmpty()) {
```

Comment:
Just set this to the boolean inside the if. And then the two ifs check the boolean. No gross if expression nonsense